lib/lolitado/db.rb

```diff
@@ -0,0 +1,98 @@
+module Lolitado
+  class Database
+
+    attr_accessor :db
+
+    def initialize db
+      @db = db
+    end
+    
+    
+    def query sql
+      result = db[sql].all
+      return result
+    end
+
+    def multiple_query sql
+      splited_sql = sql.split(';')
+      splited_sql.each do |each_sql|
+        query(each_sql)
+      end
+    end
+    
+    def query_duration sql, type = true, waiting_time = 10
+      start = Time.now
+      if type
+        result = query(sql, waiting_time)
+      else
+        result = query_empty(sql, waiting_time)
+      end
+      finish = Time.now
+      msecs = (finish - start) * 1000.0
+      hash = {'result' => result, 'duration' => msecs}
+      return hash
+    end
+
+    ## Expect to return data for the query.
+    ## If the result is empty then it will be called again until waiting_time reach.
+    def query_wait sql, waiting_time = 10
+      result = db[sql].all
+      if result.length == 0
+        if waiting_time != 0
+          sleep 1
+          result = query(sql, waiting_time - 1)
+        end
+      end
+      return result
+    end
+
+    ## Expect to return empty for the query.
+    ## If the result is not empty then it will be called again until waiting_time reach.
+    def query_empty sql, waiting_time = 10
+      result = db[sql].all
+      if result.length != 0
+        if waiting_time != 0
+          sleep 1
+          result = query(sql, waiting_time - 1)
+        end
+      end
+      return result
+    end
+
+    def insert sql
+      db[sql]
+    end
+
+    def insert_by_data data, table
+      sql = "insert into #{table} "
+      case data
+        when Array
+          data.each do |d|
+            insert_by_data(d, table)
+          end
```

Comment:
Why is there no query in this step，I'm confused.

Reply:
It's in insert_by_data method.